### ISSUE TITLE
GTK popups, that could change their size, not properly damaged

### ISSUE THREAD

**Issue by Ligthiago** [hyprwm/Hyprland#5494]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit ff114cf6f9f45b2ccc5eed95a95738b7c1a61003  ().
Date: 2024-04-06
Tag: , commits: @COMMITS@

flags: (if any)


System Information:
System name: Linux
Node name: prometheus
Release: 6.8.3
Version: #1-NixOS SMP PREEMPT_DYNAMIC Wed Apr  3 13:32:51 UTC 2024


GPU information: 
04:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Rembrandt [Radeon 680M] [1002:1681] (rev 02) (prog-if 00 [VGA controller])


os-release: ANSI_COLOR="1;34"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
BUILD_ID="24.05.20240403.fd281bd"
DOCUMENTATION_URL="https://nixos.org/learn.html"
HOME_URL="https://nixos.org/"
ID=nixos
IMAGE_ID=""
IMAGE_VERSION=""
LOGO="nix-snowflake"
NAME=NixOS
PRETTY_NAME="NixOS 24.05 (Uakari)"
SUPPORT_URL="https://nixos.org/community.html"
VERSION="24.05 (Uakari)"
VERSION_CODENAME=uakari
VERSION_ID="24.05"


plugins:

```

</details>

### Bug or Regression?

Bug

### Description

GTK popups, that can change their size has weird redrawing artifacts at bottom, see vid below. It's been there for as long as I can remember, but it's become too annoying, so I made an issue.

### How to reproduce

In case of the vid:
1) Open nautilus
2) Right-click on folder
3) Tap to upper entry, aka "open" submenu, that change popup size. See bug

It that shrimple
Basically any gtk4 popup with submenu will work, same behavior is other apps.

### Crash reports, logs, images, videos

https://github.com/hyprwm/Hyprland/assets/142721811/cc503720-1895-4a9a-b8af-822173463062

**Comment by vaxerski** [comment]
https://github.com/hyprwm/Hyprland/assets/43317083/dbc4285a-f2c2-4deb-a50e-ca2e97335514

**Comment by vaxerski** [comment]
the problem is that although I can repro on my desktop with that specific example, I can't repro in my nested session which makes debugging a pain

**Comment by Ligthiago** [comment]
Well, I can repro in nested. Maybe something with config? The only thing that fixes this for me is setting damage_tracking to full mon, obviously. In all other way bug appears.
![2024-04-08T19:23:30,064353636+04:00](https://github.com/hyprwm/Hyprland/assets/142721811/6c43c7d8-f2ea-4f58-ba78-d49b4cdff395)

**Comment by Ligthiago** [comment]
I can't reproduce this on the latest git, so I guess it's closed (until you break it again, please don't).
